### ISSUE TITLE
[Java] Checkstyle additions causing build failure in some environments

### ISSUE THREAD

**Issue by toddfarmer** [toddfarmer/arrow-migration#814]
***Note**: This issue was originally created as [ARROW-796](https://issues.apache.org/jira/browse/ARROW-796). Please see the [migration documentation](https://gist.github.com/toddfarmer/12aa88361532d21902818a6044fda4c3) for further details.*

### Original Issue Description:
Even after the conflict fixed in ARROW-677, I'm running into build problems:

```Java
SLF4J: The requested version 1.5.6 by your slf4j binding is not compatible with [1.6, 1.7]
SLF4J: See http://www.slf4j.org/codes.html#version_mismatch for further details.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Apache Arrow Java Root POM ........................ FAILURE [0.586s]
[INFO] Arrow Format ...................................... SKIPPED
[INFO] Arrow Memory ...................................... SKIPPED
[INFO] Arrow Vectors ..................................... SKIPPED
[INFO] Arrow Tools ....................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.742s
[INFO] Finished at: Sat Apr 08 17:11:40 EDT 2017
[INFO] Final Memory: 20M/633M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (validate) on project arrow-java-root: Execution validate of goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check: java.lang.AbstractMethodError: org.slf4j.impl.JDK14LoggerAdapter.log(Lorg/slf4j/Marker;Ljava/lang/String;ILjava/lang/String;[Ljava/lang/Object;Ljava/lang/Throwable;)V
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.apache.maven.plugins:maven-checkstyle-plugin:2.17
```

If I remove the checkstyle plugin from the root pom.xml, everything is OK

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-796?focusedCommentId=15961951) by Wes McKinney (wesm):*
`[~elahrvivaz]` do you know how to fix this?

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-796?focusedCommentId=15962204) by Wes McKinney (wesm):*
Update: this problem goes away if I update IntelliJ to use Maven 3.3.9 (installed separately) rather than its bundled Maven. If we don't care about this, we can close the issue

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-796?focusedCommentId=15962875) by Emilio Lahr-Vivaz (elahrvivaz):*
Yeah, I've been using 3.3.9 and didn't have a problem. I guess the readme does say 'maven 3.3 or later'. What version of maven was intellij using?

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-796?focusedCommentId=15962877) by Wes McKinney (wesm):*
IntelliJ shipped with Maven 3.0.x, so closing this as a non-issue. If others run into this problem we can refer them here (or they'll find this on Google search)

**Comment by toddfarmer** [comment]
***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-796?focusedCommentId=17585820) by @toddfarmer:*
Transitioning issue from Resolved to Closed to based on resolution field value.
